11_FFTBloom/app_resources/common.hlsl

```diff
@@ -0,0 +1,14 @@
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+
+using namespace nbl::hlsl;
+
+#define CHANNELS 3
+
+struct PushConstantData
+{
+	uint64_t colMajorBufferAddress;
+	uint64_t rowMajorBufferAddress;
+	uint32_t dataElementCount;
```

Comment:
these need comments, also I'm sure you only need one scratch buffer address at a time

Reply:
Some shaders only need one (for example first axis FFT reads from texture and write to colMajorBuffer) but some need two, for example second axis FFT: It reads from the colMajorBuffer and wants to write in row major order to keep writing coalesced, so it can't write to the same place it read from because other workgroups might not be done reading yet

In retrospective I could make both buffers be one big buffer and just have one start at an offset

Reply:
no thats fine, I just didn't realize that output writes different pattern than input.

don't change it then, just add comments (what you just wrote) to source code, esp what `dataElementcount` is